src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Windows.Forms.UITests
+{
+    public class ComboBoxTests : ControlTestBase
+    {
+        public ComboBoxTests(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        [WinFormsFact]
+        public async Task ComboBox_Select_Item_By_UpArrowKeyAsync()
+        {
+            await RunComboBoxTestAsync(async (form, comboBox) =>
+            {
+                AddItems(comboBox, 10);
+                comboBox.SelectedIndex = comboBox.Items.Count - 1;
+                int numEvents = 0;
+                comboBox.SelectedIndexChanged += (x, y) => numEvents++;
+
+                int numKeyPresses = 9;
+                for (int i = 0; i < numKeyPresses; i++)
+                {
+                    await InputSimulator.SendAsync(
+                        form,
+                        inputSimulator => inputSimulator.Keyboard.KeyPress(WindowsInput.Native.VirtualKeyCode.UP));
+
+                    Assert.Equal(i + 1, numEvents);
+                }
+
+                Assert.Equal(numKeyPresses, numEvents);
+            });
+        }
+
+        [WinFormsFact]
+        public async Task ComboBox_Select_Item_By_DownArrowKeyAsync()
+        {
+            await RunComboBoxTestAsync(async (form, comboBox) =>
+            {
+                AddItems(comboBox, 10);
+                comboBox.SelectedIndex = 0;
+                int numEvents = 0;
+                comboBox.SelectedIndexChanged += (x, y) => numEvents++;
+
+                int numKeyPresses = 9;
+                for (int i = 0; i < numKeyPresses; i++)
+                {
+                    await InputSimulator.SendAsync(
+                        form,
+                        inputSimulator => inputSimulator.Keyboard.KeyPress(WindowsInput.Native.VirtualKeyCode.DOWN));
+
+                    Assert.Equal(i + 1, numEvents);
+                }
+
+                Assert.Equal(numKeyPresses, numEvents);
+            });
+        }
+
+        private void AddItems(ComboBox comboBox, int numToAdd)
+        {
+            for (int i = 0; i < numToAdd; i++)
+            {
+                comboBox.Items.Add($"Item {i}");
+            }
+        }
+
+        private async Task RunComboBoxTestAsync(Func<Form, ComboBox, Task> runTest)
+        {
+            await RunSingleControlTestAsync(
+                testDriverAsync: runTest,
+                createControl: () =>
+                {
+                    ComboBox control = new();
+
+                    return control;
+                },
```

Comment:
I'm pretty certain this would work:

```suggestion
        private async Task RunComboBoxTestAsync(Func<Form, ComboBox, Task> runTest, int numberOfItemsToAdd, int selectedIndex)
        {
            await RunSingleControlTestAsync(
                testDriverAsync: runTest,
                createControl: () =>
                {
                    ComboBox control = new();
                    AddItems(comboBox, numberOfItemsToAdd);
                    comboBox.SelectedIndex = selectedIndex;

                    return control;
                },
```

```cs
        [WinFormsFact]
        public async Task ComboBox_Select_Item_By_DownArrowKeyAsync()
        {
            await RunComboBoxTestAsync(async (form, comboBox) =>
            {
                int numEvents = 0;
                comboBox.SelectedIndexChanged += (x, y) => numEvents++;

                // ...
            },
            numberOfItemsToAdd: 10,
            selectedIndex: 0);
        }
```

Reply:
Sorry, I forgot about one amendment.